provision/docker/provisioner.go

```diff
@@ -368,19 +368,16 @@ func (p *dockerProvisioner) Sleep(app provision.App, process string) error {
 }
 
 func (p *dockerProvisioner) Rollback(a provision.App, imageID string, evt *event.Event) (string, error) {
-	validImgs, err := image.ListValidAppImages(a.GetName())
+	imageID, err := image.GetAppImageBySuffix(a.GetName(), imageID)
 	if err != nil {
 		return "", err
 	}
-	valid := false
-	for _, img := range validImgs {
-		if img == imageID {
-			valid = true
-			break
-		}
+	imgMetaData, err := image.GetImageMetaData(imageID)
+	if err != nil {
+		return "", err
 	}
-	if !valid {
-		return "", errors.Errorf("Image %q not found in app", imageID)
+	if imgMetaData.DisableRollback {
```

Comment:
What do you think about moving this check to `func Deploy` in `app/deploy.go`? This way we wouldn't have to worry about adding this same logic to every provisioner.
Also I don't think there's need to call `GetAppImageBySuffix` here, the app package will have done this already and the imageID here would be a full valid imageID not a suffix.